packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelFactoryProvider.cs

```diff
@@ -96,15 +96,17 @@ protected override MethodProvider[] BuildMethods()
                 [
                     .. GetCollectionInitialization(signature),
                     MethodBodyStatement.EmptyLine,
-                    Return(New.Instance(typeToInstantiate.Type, [.. GetCtorParams(signature)]))
+                    Return(New.Instance(typeToInstantiate.Type, [.. GetCtorParams(signature, modelCtor.Signature)]))
                 ]);
 
                 methods.Add(new MethodProvider(signature, statements, this, docs));
             }
             return [.. methods];
         }
 
-        private static IReadOnlyList<ValueExpression> GetCtorParams(MethodSignature signature)
+        private static IReadOnlyList<ValueExpression> GetCtorParams(
```

Comment:
nit: is this getting back the args for the ctor call? If so, can we call the method GetCtorArgs?

Reply:
That's correct. I'll update the method name to better reflect that.